clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -86,7 +86,17 @@
     List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    void disconnect(String nodeId);
```

Comment:
This approach seem fine, but I am wondering if it's truly needed for KafkaAdmin. In KafkaAdminClient, we are already maintaining callsInFlight. So, if we explicit close a connection, we know all the inflight calls on that connection even if networkClient.poll() doesn't tell us, and can therefore handle them accordingly. This is how producer handles timeout as well. The slight downside of this approach is that now there are two different ways of closing a connection and the caller has to figure out which one to use. It's probably simpler to have just one way of closing a connection and be consistent whether it's the caller's responsibility to deal with inflight requests or not.